src/internal/connector/exchange/api/mail.go

```diff
@@ -357,22 +358,36 @@ func (c Mail) Serialize(
 // ---------------------------------------------------------------------------
 
 func MailInfo(msg models.Messageable) *details.ExchangeInfo {
-	sender := ""
-	subject := ptr.Val(msg.GetSubject())
-	received := ptr.Val(msg.GetReceivedDateTime())
-	created := ptr.Val(msg.GetCreatedDateTime())
-
-	if msg.GetSender() != nil &&
-		msg.GetSender().GetEmailAddress() != nil {
-		sender = ptr.Val(msg.GetSender().GetEmailAddress().GetAddress())
+	var (
+		recipient string
+		sender    = graph.UnwrapAddress(msg.GetSender())
+		subject   = ptr.Val(msg.GetSubject())
+		received  = ptr.Val(msg.GetReceivedDateTime())
+		created   = ptr.Val(msg.GetCreatedDateTime())
+		listing   = make([]string, 0)
+	)
+
+	if msg.GetToRecipients() != nil {
+		recipients := msg.GetToRecipients()
+		for _, entry := range recipients {
+			temp := graph.UnwrapAddress(entry)
+			if len(temp) > 0 {
+				listing = append(listing, temp)
+			}
+		}
+	}
+
+	if len(listing) > 0 {
+		recipient = strings.Join(listing, ",")
 	}
```

Comment:
The conditional isn't required.  A join on an empty slice produces an empty string.
```suggestion
	recipient = strings.Join(listing, ",")
```

Reply:
Actually, on that note, why reduce it to a string at all?  This should be kept as a string slice.

Reply:
Resolved within https://github.com/alcionai/corso/pull/2689

Reply:
You're still joining the slice here.

Reply:
It will be removed!